qri/dataset.py

```diff
@@ -60,48 +59,76 @@ def __repr__(self):
         return 'Commit(%s)' % r
 
 
+def is_from_list(obj):
+    version_info_fields = ['bodySize', 'bodyRows', 'bodyFormat', 'numErrors', 'commitTime']
+    return bool(set(obj) & set(version_info_fields))
+
+
 class Dataset(object):
     def __init__(self, obj):
         # Fields that are always present
         set_fields(self, obj, ['username', 'name', 'profileID', 'path'])
         if self.username is None and 'peername' in obj:
             self.username = obj.get('peername')
-        # Fields returned by `get` commands
-        set_fields(self, obj, ['bodyPath', 'previousPath'])
-        # Version info
-        self.versionInfo = self._build_version_info(obj)
+
+        self._is_populated = False
+        if not is_from_list(obj):
+            self._populate(obj)
+
+    def _ensure_populated(self):
+        if self._is_populated:
+            return
+        ref = dsref.Ref(self.username, self.name)
+        self._populate(loader.instance().get_dataset_object(ref))
+
+
+    # TODO - write tests
+    def _populate(self, obj):
+        self.body_path_value = obj.get('bodyPath')
+        self.previous_path_value = obj.get('previousPath')
+
         # Subcomponents
         self.commit_component = Commit(obj.get('commit'))
         self.meta_component = Meta(obj.get('meta'))
         self.readme_component = Readme(obj.get('readme'))
         self.structure_component = Structure(obj.get('structure'))
         self.body_component = None
 
-    def _build_version_info(self, obj):
-        info = version_info.VersionInfo(obj)
-        # TODO(dustmop): Remove me after this typo is fixed in qri core
-        if 'bodyFromat' in obj:
```

Comment:
The relevant fix looks like it was committed almost 2 months ago. I figured it's time to take this out? Or is there a lag in releases?

Reply:
This should be safe to remove now, yes.